syntax = "proto3";

package envoy.extensions.pattern_template.v3;

import "udpa/annotations/status.proto";

option java_package = "io.envoyproxy.envoy.extensions.pattern_template.v3";
option java_outer_classname = "PatternTemplateRewriteProto";
option java_multiple_files = true;
option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/pattern_template/v3;pattern_templatev3";
option (udpa.annotations.file_status).package_version_status = ACTIVE;

// [#protodoc-title: Pattern matcher]
// [#extension: envoy.pattern_template]

// Holds the segments for rewriting urls base on pattern templates
message PatternTemplateRewrite {
  message RewriteSegment {
    oneof segment_type {
      // Represents a segment of the rewritten URL, including any path segments,
      // slash and prefix.
      string literal = 1;

      // Represents an index into the RE2 capture which value should be used
      // to construct the rewritten URL. Note that the index should be greater
      // than 0 as 0 index into the whole match RE2 pattern.
      int32 var_index = 2;
    }
  }

  repeated RewriteSegment segments = 3;
}
